tasks/main.yml

```diff
@@ -1,36 +1,107 @@
 ---
-- name: Ensure directory exists
+- name: Ensure group for backups user exists
+  group:
+    name: "{{ backups_role_user_group }}"
+    state: present
+
+- name: Ensure restricted user for backups exists
+  user:
+    name: "{{ backups_role_user_name }}"
+    group: "{{ backups_role_user_group }}"
+    system: true
+    state: present
+  when: backups_role_user_name != 'root'
+
+
+- name: Ensure directory for backup script exists
   file:
-    path: "{{ backups_role_path }}"
+    path: "{{ backups_role_script_path }}"
     state: directory
-    owner: "{{ backups_role_user }}"
-    group: "{{ backups_role_group }}"
+    owner: "{{ backups_role_user_name }}"
+    group: "{{ backups_role_user_group }}"
     mode: 0775
 
 - name: Copy backup script
   template:
     src: backups.sh.j2
-    dest: "{{ backups_role_path }}/backups.sh"
-    owner: "{{ backups_role_user }}"
-    group: "{{ backups_role_group }}"
+    dest: "{{ backups_role_script_path }}/{{ backups_role_script_name }}"
+    owner: "{{ backups_role_user_name }}"
+    group: "{{ backups_role_user_group }}"
     mode: 0775
 
+- name: Ensure directory for backups temporary data exists
+  file:
+    path: "{{ backups_role_tmp_path }}"
+    state: directory
+    owner: "{{ backups_role_user_name }}"
+    group: "{{ backups_role_user_group }}"
+    mode: 0775
+
+- name: Ensure postgresql "role" for backups exists with limited access to db
+  postgresql_user:
+    db: "{{ backups_role_db_name }}"
+    name: "{{ backups_role_postgresql_user_name }}"
+    password: "{{ backups_role_postgresql_user_password }}"
+    # This provides '{{ name  }}=c/{{ become_user }}' in {{ db }} database
+    # Plus any other access it can have through privileges granted to PUBLIC
+    # This `priv` key is not flexible enough to grant SELECT priv on *any* table
+    # inside database. Thus, we do it in two steps.
+    priv: "CONNECT"
+  become: true
+  become_user: "{{ postgresql_user }}"
+
+- name: Ensure postgresql "role" for backups has "select" access to any table in {{ backups_role_db_name }}
+  postgresql_privs:
+    db: "{{ backups_role_db_name }}"
+    role: "{{ backups_role_postgresql_user_name }}"
+    objs: ALL_IN_SCHEMA
+    # This ensures that this priv exists: {{ role }}=r/{{ become_user }}
+    privs: SELECT
+  become: true
+  become_user: "{{ postgresql_user }}"
+
+- name: Ensure postgresql "role" for backups has no other access to any table in {{ backups_role_db_name }}
+  postgresql_privs:
+    db: "{{ backups_role_db_name }}"
+    role: "{{ backups_role_postgresql_user_name }}"
+    objs: ALL_IN_SCHEMA
+    # Revoke all privs except SELECT
+    # https://www.postgresql.org/docs/current/sql-grant.html
+    privs: "INSERT,UPDATE,DELETE,TRUNCATE,REFERENCES,TRIGGER"
+    state: absent
+  become: true
+  become_user: "{{ postgresql_user }}"
+
+- name: Let backup user to use `{{ backups_role_sudoers_cmd_pattern }}` as root
+  lineinfile:
+    path: /etc/sudoers
```

Comment:
A cleaner way to do this is to use `sudoers.d` instead of touching `/etc/sudoers` file.

Example: https://github.com/coopdevs/timeoverflow-provisioning/blob/master/roles/sys_admins/tasks/main.yml#L25

Reply:
Totally. Let's do it this way. I wasn't that comfortable editing sudoers inline.

Reply:
Addressed in 0c4a8dbbea5ca9bf73328374b0f1210ecf5213c1